src/KdbOptions.h

```diff
@@ -162,6 +162,12 @@ class KdbOptions
 
     K keys = kK( kK( dict )[index] )[0];
     K values = kK( kK( dict )[index] )[1];
+    if( KS != keys->t ){
+        throw InvalidOption( "Unsupported KDB data type for NULL_MAPPING keys" );

```

Comment:
It's a minor thing, but its handy for the user to report the expected type
Look at `KdbOptions::KdbOptions()`